GoogleSignIn/Tests/Unit/GIDVerifyAccountDetailTest.m

```diff
@@ -0,0 +1,110 @@
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h"
+
+#import "GoogleSignIn/Tests/Unit/GIDFakeMainBundle.h"
+
+static NSString * const kClientId = @"FakeClientID";
+
+@interface GIDVerifyAccountDetailTests : XCTestCase {
+@private
+  // The |UIViewController| object being tested.
+  UIViewController *_presentingViewController;
+
+  // Fake [NSBundle mainBundle].
+  GIDFakeMainBundle *_fakeMainBundle;
+
+  // The |GIDVerifyAccountDetail| object being tested.
+  GIDVerifyAccountDetail *_verifyAccountDetail;
+
+  // The list of account details when testing [GIDVerifiableAccountDetail].
+  NSArray<GIDVerifiableAccountDetail *> *_verifiableAccountDetails;
+}
+@end
+
+@implementation GIDVerifyAccountDetailTests
+
+#pragma mark - Lifecycle
+
+- (void)setUp {
+  [super setUp];
+
+  _presentingViewController = [[UIViewController alloc] init];
+
+  _verifyAccountDetail = [[GIDVerifyAccountDetail alloc] init];
+
+  GIDVerifiableAccountDetail *ageOver18Detail = [[GIDVerifiableAccountDetail alloc] initWithAccountDetailType:GIDAccountDetailTypeAgeOver18];
+  _verifiableAccountDetails = @[ageOver18Detail];
+
+  _fakeMainBundle = [[GIDFakeMainBundle alloc] init];
+  [_fakeMainBundle startFakingWithClientID:kClientId];
+  [_fakeMainBundle fakeAllSchemesSupported];
+}
+
+
+#pragma mark - Tests
+
+- (void)testPresentingViewControllerException {
+  _presentingViewController = nil;
+
+  XCTAssertThrows([_verifyAccountDetail verifyAccountDetails:_verifiableAccountDetails
+                                    presentingViewController:_presentingViewController
+                                                  completion:nil]);
+}
+
+- (void)testClientIDMissingException {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnonnull"
+  _verifyAccountDetail.configuration = [[GIDConfiguration alloc] initWithClientID:nil];
+#pragma GCC diagnostic pop
+  BOOL threw = NO;
+  @try {
+    [_verifyAccountDetail verifyAccountDetails:_verifiableAccountDetails
+                      presentingViewController:_presentingViewController
+                                    completion:nil];
+  } @catch (NSException *exception) {
+    threw = YES;
+    XCTAssertEqualObjects(exception.description,
+                          @"You must specify |clientID| in |GIDConfiguration|");
+  } @finally {
+  }
+  XCTAssert(threw);
+}
+
+- (void)testSchemesNotSupportedException {
+  [_fakeMainBundle fakeMissingAllSchemes];
+  BOOL threw = NO;
+  @try {
+    [_verifyAccountDetail verifyAccountDetails:_verifiableAccountDetails
+                      presentingViewController:_presentingViewController
+                                    completion:nil];
+  } @catch (NSException *exception) {
+    threw = YES;
+    XCTAssertEqualObjects(exception.description,
+                          @"Your app is missing support for the following URL schemes: "
```

Comment:
The reason description should be a literal. I'm asking that you make the second comparison string be a constant in this file.

Reply:
Gotcha, I think creating the constant is no longer relevant since the assert method has switched to `XCTAssertThrowsSpecificNamed` and wouldn't use the constant, but let me know if I misunderstood.

Reply:
Agreed. Resolving.